e2e: add tests for static-pools

askervin opened the pull request:
Opening the first static-pools tests patch for comments & discussion:
- What would be the minimum features that you'd like to see tested as part of this commit?
- Do you see major issues with the test structure? For instance, unlike in memtier tests, here cri-resmgr, agent and webhook are restarted before every test execution.

askervin commented:
This PR depends on PR's #461, #463, #465, #466, #467, #468, #469, #470 and #471.

klihub commented:
I'll leave this to @marquiz ...

askervin commented:
@marquiz, @klihub, all tests now pass when PR #465 is picked.

I think this is patch is now ready, "RFC" cleared.

Ran all tests also rebased on top of PR #494 and VM deleted, and all passed there as well.

klihub commented on test/e2e/policies.test-suite/static-pools/static-pools-lib.source.sh:
```diff
@@ -0,0 +1,25 @@
+    local webhook_running=0
+    out "# Relaunching cri-resmgr and agent, launch webhook if not already running"
+
+    vm-command-q "kubectl get mutatingwebhookconfiguration/cri-resmgr" >/dev/null 2>&1 && {
```
extreme nitpick: since we assume bash, isn't `>& /dev/null` an easier/more readable way to write this ?

askervin replied:
Thanks, fixed!

klihub commented:
I skimmed through this and the test scripts et. al look good to me. I don't know enough about `CMK/static-pools` to tell whether the test set coverage is good enough. I let @marquiz pass judgement on that.

askervin commented:
Latest force-push change in addition to `s|>/dev/null 2>&1|>&/dev/null`: switched from `vm-command` to `vm-run-until` in the negative tests where checking error from pod description. It's more robust. Fixed a false test failure.

klihub commented:
Looks good to me.

klihub commented:
WDYT @marquiz ?

marquiz reviewed (approved):
Wow, really nice work @askervin ! And, apologies for the delay in my review.

The tests look really extensive. There are just two small things that popped into my mind:
- we could add check that the `CMK_CPUS_INFRA` and `CMK_CPUS_SHARED` env variables are correctly set
- we could have a negative test case for exclusive pool where no exclusive cores is requested

But these can be left as a future excercise (add an issue for those). I'm fine with merging this in its current state, too.

askervin commented:
@marquiz , @klihub , thanks for the review!

Because this inconveniently large patch looks approved, I added tests that @marquiz mentioned to PR #496  - in order to let this patch flow in now.

However, that PR heavily depends on this patch, so now the only commit in this patch is included in that PR, too. I don't know if this is very convenient way to handle this kind of dependency, but could not really come up with anything better either...

askervin commented:
Rebased and squashed #496 into this.

```
$ ./run_tests.sh policies.test-suite/static-pools/
...
PASS static-pools/n4c16/test00-node-status
PASS static-pools/n4c16/test01-exclusive-pods
PASS static-pools/n4c16/test02-pods-without-cmk
PASS static-pools/n4c16/test03-cmk-isolate
PASS static-pools/n4c16/test04-cmk-isolate-noaffinity
PASS static-pools/n4c16/test05-negative-tests
```

marquiz reviewed (approved):
LGTM!
